whitelabelco/whitelabel-components
# feat(button): adds Button component, integrates theme

Link to Issue/Ticket -->
Resolves: Issue #16 

## Description
This PR:
- Adds Button component with basic functionality and defined types.
- Adds Button styling, test, and complete stories to showcase functionality.
- Integrates and adjusts theme system to dynamically work with components like Button.
- Adds global theme to preview to allow for visual toggle of themes.
- Removes unnecessary files like globalStyling (might add again later)

## To Test :mag: 
- `yarn install && yarn build-storybook` for setup if necessary
- Go to Button atom on storybook and test options of props both on canvas and in docs
- Verify no accessibility warnings are showing


## Checklist :white_check_mark:
- [ ] I have re-assigned the Github ticket and moved the ticket on the project board.
- [ ] If I have added an ENV var or other secrets, I have updated Dashlane and the ReadMe (and any other applicable docs)

## Additional Details and Screenshots :art: 
- Types were a big hurtle while building this component. My thought process for this went through many iterations, and separating and defining types so extensively is what allowed for a dynamic and organized theme system.
- Button styling was clumped together to work with twin.macro and keep themes mostly uniform. Colors used in the variants are more like palettes to follow, consisting of background color, text color, and hover color. I figured instead of adding more props, (there's already a lot) and adding the potential for the user to create a poorly implemented palette like choosing an inaccessible text color on an incompatible background, there would be more opinionated options. This extends to some of the other stylings in theme too.

<img width="779" alt="Screenshot 2023-04-18 at 9 16 42 AM" src="https://user-images.githubusercontent.com/81177655/232827633-7078259c-2b20-433c-a515-c801233754c8.png">
<img width="871" alt="Screenshot 2023-04-18 at 9 36 46 AM" src="https://user-images.githubusercontent.com/81177655/232828932-d71d68c7-56db-461f-89ec-49c938709e2f.png">


- The button with an icon is still WIP, as I'm building out the media component for it now. As I see it currently, the size of the icon would be determined in the button, preferably in relation to the text size. Icon color might still be an issue, but that could always be solved with multiple icon files, or overlays.
- We decided against hero icons since each icon is a component itself, which makes it difficult to work with. We could potentially add an icon library later, like fontAwesome, but it was determined not vital for mvp. Totally fine if we should discuss that choice more.
- I chose the props I did in the hopes of minimizing the need for custom styling down the line by covering most basic stylings for a button but not going too overboard with props and details. The button allows for additional props to be added as long as they're props you'd find on a button html tag like style or accessibility props. If a button is passed a child, it sheds all built in styling and inherits the styling of the child to allow for an onClick wrapper option for other components. 

<img width="1010" alt="Screenshot 2023-04-18 at 9 39 51 AM" src="https://user-images.githubusercontent.com/81177655/232829679-465def80-0e6b-4fd7-849b-77136b7d10be.png">

<img width="821" alt="Screenshot 2023-04-18 at 9 40 34 AM" src="https://user-images.githubusercontent.com/81177655/232829863-329b2a17-176d-4142-8eda-9bca79a86143.png">

- Left to do:
- Build media component and integrate it into iconButton / iconOnly options for button
- (for text component later) add typography in theme
- potentially refactor to simplify if possible
